app/code/Magento/User/Controller/Adminhtml/Auth/Forgotpassword.php

```diff
@@ -142,4 +144,22 @@ public function execute()
         $this->_view->loadLayout();
         $this->_view->renderLayout();
     }
+
+    /**
+     * Retrieve and validate existing password reset token; returns true on
+     * success, false on failure
+     *
+     * @param User $user
+     * @return bool
+     */
+    protected function userHasValidPasswordResetToken(User $user): bool
+    {
+        try {
+            $newPassResetToken = $user->getRpToken();
+            $this->_validateResetPasswordLinkToken((int) $user->getId(), $newPassResetToken);
+            return true;
+        } catch (LocalizedException $exception) {
+            return false;
```

Comment:
The same here, I think there might be different cases when LocalizedException  might be thrown and it might be some other issue, not that we expected

Reply:
`LocalizedException` is explicitly thrown from `_validateResetPasswordLinkToken()` to indicate failure. Should I check for the error message here and re-throw if not one of the three expected strings? Or perhaps change `_validateResetPasswordLinkToken()` to throw a more specific / different exception that we can catch here?

<details><summary>Magento\User\Controller\Adminhtml\Auth::_validateResetPasswordLinkToken()</summary>

https://github.com/magento/magento2/blob/8653503c243de1db59d9bc38356d744bf83e51e8/app/code/Magento/User/Controller/Adminhtml/Auth.php#L49-L72

</details>

Reply:
Could you just add comment about that this type of exception will be thrown here? I think it will be enough.